noxfile.py

```diff
@@ -4,10 +4,96 @@
 
 nox.options.default_venv_backend = "uv"
 
+supported_python_versions = ["3.10", "3.11", "3.12"]
 
-maxpython = "3.12"
+maxpython = max(supported_python_versions)
+minpython = min(supported_python_versions)
 
-doc_requirements = ("-r", "ci_requirements/requirements_docs_py312.txt")
+requirements_directory = "ci_requirements"
+
+
+def get_requirements_file(
```

Comment:
One of the benefits of switching to nox is that we can define functions to help create testing environments!